### PR TITLE
iteration4

### PR DESCRIPTION
**Pull request by alinazaitseva1**

### CODE REVIEW

File: DataStructureMemories/DataStructureMemories/ControllerDataStructure/DataStructureTableViewController.swift [hunk 1]
```diff
@@ -21,20 +21,30 @@ class DataStructureTableViewController: UIViewController, UITableViewDataSource,
...
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
         print(indexPath.row)
+        
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let detailViewController = storyboard.instantiateViewController(withIdentifier:  "DetailViewController") as! DetailViewController
+        
+        detailViewController.titleStructure = dataStructureManipulatios.getValueInDataSource(item: indexPath.row)
+        detailViewController.detailDescription = dataStructureManipulatios.descriptOfDataStructure[indexPath.row]
+
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/ControllerDataStructure/DataStructureTableViewController.swift]: Why don't we pass whole model to the Detail controller

File: DataStructureMemories/DataStructureMemories/ControllerDataStructure/DetailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
+    
+    @IBAction func buttonShowText(_ sender: Any) {
+        heightConstraint.priority = UILayoutPriority(rawValue: 250)
+        gradientView.isHidden = true
+        view.layoutIfNeeded()
+    }
+}
+extension UIView {
+    func opacityGradient() {
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/ControllerDataStructure/DetailViewController.swift]: cool!

File: DataStructureMemories/DataStructureMemories/ModelDataStructure/DataStructEntity.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//
+
+import Foundation
+
+class DataStructEntity: DataStructMemory {
+    
+    private let title: String
+    private let descr: String
+    
+    func returnTitleOfString() -> String {
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/ModelDataStructure/DataStructEntity.swift]: please remove return word from the function name

File: DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemory.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Created by Alina Zaitseva on 4/17/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+
+import Foundation
+
+protocol DataStructMemory {
+    
+    func returnTitleOfString() -> String
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemory.swift]: please remove "return" word from the function name

File: NewProject/NewProject/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,46 @@
+//
+//  ViewController.swift
```
Comment (AlexeyTyurenkov) [inline NewProject/NewProject/ViewController.swift:2]: please remove unused files from the code base

File: DataStructureMemories/DataStructureMemories/ControllerDataStructure/DetailViewController.swift [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        self.title = titleStructure ?? "There is no such title"
+        self.descriptionValue.text = detailDescription ?? "There is no such description"
+        gradientView.opacityGradient()
+    }
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    @IBOutlet weak var gradientView: UIView!
+    var gradientLayer: CAGradientLayer!
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/ControllerDataStructure/DetailViewController.swift:29]: i'm not sure if it should be forced unwrapped optional

**Review by AlexeyTyurenkov — CHANGES_REQUESTED** [review]

File: NewProject/NewProject/ViewController.swift (hunk 5 above)
Comment (alinazaitseva1) [reply NewProject/NewProject/ViewController.swift:2]: sure

File: DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemory.swift (hunk 4 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemory.swift]: got it

File: DataStructureMemories/DataStructureMemories/ControllerDataStructure/DetailViewController.swift (hunk 6 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/ControllerDataStructure/DetailViewController.swift:29]: after I delete force unwrapped optional it start use unpleasant words)

File: DataStructureMemories/DataStructureMemories/ControllerDataStructure/DataStructureTableViewController.swift (hunk 1 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/ControllerDataStructure/DataStructureTableViewController.swift]: fixed it

**Pull request merged**